pinot-query-planner/src/main/java/org/apache/pinot/query/QueryEnvironment.java

```diff
@@ -79,11 +80,14 @@ public class QueryEnvironment {
   // Pinot extensions
   private final Collection<RelOptRule> _logicalRuleSet;
   private final WorkerManager _workerManager;
+  private final TableCache _tableCache;
```

Comment:
passing in _tableCache seems to be a bit weird to me. but i don't have a better abstraction here. 
can we add a TODO and indicate exactly what this tableCache is intended to be used and ensure that this is not being abused. 

b/c all info that planner needs are already encapsulated inside the CalciteSchema.

Reply:
actually IMO this info should be encapsulated in `org.apache.pinot.query.catalog.PinotTable` but we don't have to fix this in this PR